catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -115,63 +118,104 @@
 
   private boolean storeOriginalImage = true;
 
+  private CatalogFramework catalogFramework;
+
   static {
     IIORegistry.getDefaultInstance().registerServiceProvider(new J2KImageReaderSpi());
   }
 
   private double maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
 
   @Override
-  public CreateStorageRequest process(CreateStorageRequest createStorageRequest)
-      throws PluginExecutionException {
-    if (createStorageRequest == null) {
-      throw new PluginExecutionException(
-          "process(): argument 'createStorageRequest' may not be null.");
+  public CreateResponse process(CreateResponse createResponse) throws PluginExecutionException {
+    if (createResponse == null) {
+      throw new PluginExecutionException("process(): argument 'createResponse' may not be null.");
     }
-
-    process(createStorageRequest.getContentItems());
-    return createStorageRequest;
+    updateContent(
+        createResponse.getCreatedMetacards(), createResponse.getRequest().getProperties());
+    return createResponse;
   }
 
   @Override
-  public UpdateStorageRequest process(UpdateStorageRequest updateStorageRequest)
-      throws PluginExecutionException {
-    if (updateStorageRequest == null) {
-      throw new PluginExecutionException(
-          "process(): argument 'updateStorageRequest' may not be null.");
+  public UpdateResponse process(UpdateResponse updateResponse) throws PluginExecutionException {
+    if (updateResponse == null) {
+      throw new PluginExecutionException("process(): argument 'updateResponse' may not be null.");
     }
+    updateContent(
+        updateResponse
+            .getUpdatedMetacards()
+            .stream()
+            .map(Update::getNewMetacard)
+            .collect(Collectors.toList()),
+        updateResponse.getRequest().getProperties());
+    return updateResponse;
+  }
 
-    process(updateStorageRequest.getContentItems());
-    return updateStorageRequest;
+  @Override
+  public DeleteResponse process(DeleteResponse deleteResponse) throws PluginExecutionException {
+    return deleteResponse;
   }
 
-  private boolean isNitfMimeType(String rawMimeType) {
-    try {
-      return NITF_MIME_TYPE.match(rawMimeType);
-    } catch (MimeTypeParseException e) {
-      LOGGER.debug("unable to compare mime types: {} vs {}", NITF_MIME_TYPE, rawMimeType);
+  private void updateContent(List<Metacard> metacards, Map<String, Serializable> properties) {
```

Comment:
Is there a chance that this list of metacards could contain duplicates? Maybe this should be a `Set` instead so we don't generate the images more than once.

Reply:
Don't think there should be any duplicates here but can switch to a set.